packages/carbon-web-components/src/components/side-panel/side-panel-story.ts

```diff
@@ -0,0 +1,423 @@
+          kind=${BUTTON_KIND.GHOST}
+          size="sm"
+          tooltip-text="Delete">
+          ${Trashcan({ slot: 'icon' })}
+        </cds-button>
+      `;
+    default:
+      return null;
+  }
+};
+
+const actionItems = {
+  'No actions': 0,
+  'One button': 1,
+  'Two buttons with ghost': 2,
+  'Two buttons with danger': 3,
+  'Three buttons with ghost': 4,
+  'Three buttons with danger': 5,
+};
+
+// TODO: There are problems switching this
+const getActionItems = (index) => {
+  switch (index) {
+    case 1:
+      return html`<cds-button key="p" slot="actions" kind=${BUTTON_KIND.PRIMARY}
+        >Primary</cds-button
+      >`;
+    case 2:
+      return html`
+        <cds-button slot="actions" kind=${BUTTON_KIND.GHOST}>Ghost</cds-button>
+        <cds-button slot="actions" kind=${BUTTON_KIND.PRIMARY}
+          >Primary</cds-button
+        >
+      `;
+    case 3:
+      return html` <cds-button slot="actions" kind=${BUTTON_KIND.DANGER}
+          >Danger</cds-button
+        >
+        <cds-button slot="actions" kind=${BUTTON_KIND.PRIMARY}
+          >Primary</cds-button
+        >`;
+    case 4:
+      return html` <cds-button slot="actions" kind=${BUTTON_KIND.GHOST}
+          >Ghost</cds-button
+        >
+        <cds-button slot="actions" kind=${BUTTON_KIND.SECONDARY}
+          >Secondary</cds-button
+        >
+        <cds-button slot="actions" kind=${BUTTON_KIND.PRIMARY}
+          >Primary</cds-button
+        >`;
+    case 5:
+      return html`<cds-button
+          key="danger"
+          slot="actions"
+          kind=${BUTTON_KIND.DANGER}
+          >Danger</cds-button
+        >
+        <cds-button key="secondary" slot="actions" kind=${BUTTON_KIND.SECONDARY}
+          >Secondary</cds-button
+        >
+        <cds-button key="primary" slot="actions" kind=${BUTTON_KIND.PRIMARY}
+          >Primary</cds-button
+        >`;
+    default:
+      return null;
+  }
+};
+
+export default {
+  title: 'Components/SidePanel',
+  decorators: [(story) => html` ${story()} `],
+  parameters: {
+    ...storyDocs.parameters,
+  },
+};
+
+const DefaultTemplate = (args) => {
+  const {
+    actionItems,
+    actionToolbarItems,
+    content,
+    label,
+    open,
+    containerClass,
+    placement,
+    preventCloseOnClickOutside,
+    selectorPageContent,
+    size,
+    slideIn,
+    subtitle,
+  } = args?.['cds-side-panel'] ?? {};
+
+  const actions = [
+    { kind: BUTTON_KIND.GHOST, label: 'Ghost' },
+    { kind: BUTTON_KIND.SECONDARY, label: 'Secondary' },
+    { kind: BUTTON_KIND.PRIMARY, label: 'Primary' },
+  ];
+  let actionsStacked = false;
+
+  CDSSidePanel.sortActions(actions, actionsStacked);
+
+  const handleStackingChange = ({ detail }) => {
+    actionsStacked = !actionsStacked; // detail.stacked;
+    CDSSidePanel.sortActions(actions, actionsStacked);
+  };
+
+  return html`
+    <div
+      style="position: fixed; top: 0; left: 0; display: grid; grid-template-rows: 3rem 1fr; width: 100vw; height: 100vh;">
```

Comment:
can we move these inline styles to the `story-styles.scss` file instead?

Reply:
Done